EchoPro/survey.py

```diff
@@ -36,741 +23,440 @@ class Survey:
-
-        semi_vario = SemiVariogram(
-            krig_mesh.transformed_transect_df.x_transect.values,
-            krig_mesh.transformed_transect_df.y_transect.values,
-            self.bio_calc.transect_results_gdf[
-                "biomass_density_adult"
-            ].values.flatten(),
-            params["lag_res"],
-            params["nlag"],
-        )
-
-        return semi_vario
-
-    def get_kriging(self, params: krig_param_type) -> Kriging:
-        """
-        Initializes a ``Kriging`` object using the
-        provided parameters
-
-        Parameters
-        ----------
-        params : dict
-            Kriging specific parameters. Contains the following parameters:
-
-            - ``k_max: int`` -- the maximum number of data points within the
-              search radius.
-            - ``k_min: int`` -- the minimum number of data points within the
-              search radius.
-            - ``R: float`` -- search radius for Kriging
-            - ``ratio: float`` -- acceptable ratio for the singular values
-              divided by the largest singular value.
-            - ``s_v_params: dict`` -- dictionary specifying the parameter values
-              for the semi-variogram model.
-            - ``s_v_model: Callable`` -- a Semi-variogram model from the ``SemiVariogram`` class
-
-        Returns
-        -------
-        krig : Kriging
-            An initialized ``Kriging`` object that provides users with
-            access to routines that run Ordinary Kriging and routines
-            that plot final Kriging results
-        """
-
-        if not params:
-            raise ValueError("You must provide parameters for the Kriging routine!")
-
-        # make sure all parameters are included
-        if set(krig_type_dict.keys()).difference(set(params.keys())):
-            raise ValueError("Some required parameters where not provided!")
-
-        # check that all types are correct for params
-        for key, val in params.items():
-            expected_type = krig_type_dict.get(key)
-            if not isinstance(val, expected_type):
-                raise TypeError(
-                    f"The Kriging parameter {key} is not of type {expected_type}"
-                )
-
-        krig = Kriging(
-            self,
-            params["k_max"],
-            params["k_min"],
-            params["R"],
-            params["ratio"],
-            params["s_v_params"],
-            params["s_v_model"],
-        )
-
-        return krig
-
-    def compute_kriging_results(self, kriging_params: dict) -> None:
+            attribute_name = LAYER_NAME_MAP.get(config_map[0])['superlayer'][0]
+            layer_name = LAYER_NAME_MAP.get(config_map[0])['name']
+            internal = getattr( self , attribute_name )
+
+            # Add original data layer (e.g. stratification, kriging, etc.) if needed
+            if config_map[0] not in internal.keys():
+                internal = internal.setdefault(layer_name, {})
```

Comment:
This condition should never happen, since these should be initialized already in `__init__` (see https://github.com/uw-echospace/EchoPro/pull/161/files?diff=unified&w=0#r1449843433).

Reply:
This happens for cases such as `self.acoustics` that is initialized since it will also store other acoustic information (e.g. `sigma_bs` besides `nasc`). Consequently, the outermost layer of the config tree in the YAML file is `NASC` instead of acoustics. Therefore, it is missing. I suppose this could be initialized via: 
```
self.acoustics = {
    'nasc_df': pd.DataFrame() 
}

self.statistics = {
    'kriging: {
        'mesh_df': pd.DataFrame() ,
        'isobath_200m_df': pd.DataFrame() ,
        'vario_krig_para_df': pd.DataFrame()
    }
}
```

and so on. I'll briefly play around with this, or simply add this structure to the API (probably the easiest option).

Reply:
Might be useful to initialize the dicts with the structure you anticipate, so that the intention is clear, and errors are thrown automatically.